spec.html

```diff
@@ -560,14 +560,89 @@ <h1>get %TypedArray%.prototype.length</h1>
       <p>This function is not generic. The *this* value must be an object with a [[TypedArrayName]] internal slot.</p>
     </emu-clause>
 
+    <emu-clause id="sec-%typedarray%.prototype.copywithin">
+      <h1>%TypedArray%.prototype.copyWithin ( _target_, _start_ [ , _end_ ] )</h1>
+      <p>The interpretation and use of the arguments of %TypedArray%`.prototype.copyWithin` are the same as for `Array.prototype.copyWithin` as defined in <emu-xref href="#sec-array.prototype.copywithin"></emu-xref>.</p>
+      <p>When the `copyWithin` method is called, the following steps are taken:</p>
+      <emu-alg>
+        1. Let _O_ be the *this* value.
+        1. Perform ? ValidateTypedArray(_O_).
+        1. <ins>Let _getBufferByteLength_ be MakeIdempotentArrayBufferByteLengthGetter(~SeqCst~).</ins>
+        1. Let _len_ be <del>_O_.[[ArrayLength]]</del><ins>IntegerIndexedObjectLength(_O_, _getBufferByteLength_).</ins>.
+        1. <ins>Assert: _len_ is not ~out-of-bounds~.</ins>
+        1. Let _relativeTarget_ be ? ToIntegerOrInfinity(_target_).
+        1. If _relativeTarget_ is -&infin;, let _to_ be 0.
+        1. Else if _relativeTarget_ &lt; 0, let _to_ be max(_len_ + _relativeTarget_, 0).
+        1. Else, let _to_ be min(_relativeTarget_, _len_).
+        1. Let _relativeStart_ be ? ToIntegerOrInfinity(_start_).
+        1. If _relativeStart_ is -&infin;, let _from_ be 0.
+        1. Else if _relativeStart_ &lt; 0, let _from_ be max(_len_ + _relativeStart_, 0).
+        1. Else, let _from_ be min(_relativeStart_, _len_).
+        1. If _end_ is *undefined*, let _relativeEnd_ be _len_; else let _relativeEnd_ be ? ToIntegerOrInfinity(_end_).
+        1. If _relativeEnd_ is -&infin;, let _final_ be 0.
+        1. Else if _relativeEnd_ &lt; 0, let _final_ be max(_len_ + _relativeEnd_, 0).
+        1. Else, let _final_ be min(_relativeEnd_, _len_).
+        1. Let _count_ be min(_final_ - _from_, _len_ - _to_).
+        1. If _count_ &gt; 0, then
+          1. NOTE: The copying must be performed in a manner that preserves the bit-level encoding of the source data.
+          1. Let _buffer_ be _O_.[[ViewedArrayBuffer]].
+          1. <ins>Let _regetBufferByteLength_ to MakeIdempotentArrayBufferByteLengthGetter(~SeqCst~).</ins>
```

Comment:
Small typo here and in "slice":

```diff
-1. <ins>Let _regetBufferByteLength_ to MakeIdempotentArrayBufferByteLengthGetter(~SeqCst~).</ins>
+1. <ins>Let _regetBufferByteLength_ be MakeIdempotentArrayBufferByteLengthGetter(~SeqCst~).</ins>
```

...though I wonder what you think about re-assigning "getBufferByteLength" instead. While a distinct variable highlights the consideration we're making, it also insinuates that there's further need for the original closure. As I understand it, we do not need or want the original value of "getBufferByteLength" at this point, so over-writing it would be more clear from that perspective. We could use an informative note to underscore the consideration and also explain it a bit.

Reply:
It is a bit confusing, but the original closure could still be used. The reason this idempotent getter exists is to avoid multiple shared memory loads of the length, because each such load is always observable. So the getter loads it once and caches it. Once cached, you could re-use the getter. It's... not a great solution, and probably warrants a bigger refactor when we're getting ready for Stage 4. I'd like to discuss with the other editors what's a clearer way here.

Reply:
My suggestion is editorially-motivated. While the original closure could still be used, I can't think of a case where the cached value would be pertinent. By storing the new closure in a new variable, the text suggests that such a case exists, and it presents a hazard for future editors who may mistakenly use "getBufferByteLength" instead of "regetBufferByteLength".

Reply:
Fair enough, but I think the bigger problem is that the idempotent getter thing is just not a very clear mechanism to begin with. Happy to change it here to setting.

Reply:
Yeah, if you expect this will be refactored, then I agree that this isn't so important. Though we still ought to correct the grammar ("to" to "be").